### PR TITLE
cargo-chef: init at 0.1.50

### PR DESCRIPTION
**Pull request by kkharji**
###### Description of changes
Add cargo chef as rust package

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
Please check CONTRIBUTING.md for our contributing guidelines

File: pkgs/development/tools/rust/cargo-chef/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, stdenv, fetchurl, runCommand, fetchCrate, rustPlatform, Security, openssl
```
Comment (Artturin) [inline pkgs/development/tools/rust/cargo-chef/default.nix:1]: This file isn't called anywhere and also most of these inputs aren't being used

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -15917,6 +15917,12 @@
     github = "kuwii";
     githubId = 10705175;
   };
+  kkharji = {
```
Comment (figsoda) [inline maintainers/maintainer-list.nix:15920]: please put this into a separate commit `maintainers: add kkharji`

File: pkgs/development/tools/rust/cargo-chef/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{ lib, fetchCrate, rustPlatform, }:
```
Comment (figsoda) [inline pkgs/development/tools/rust/cargo-chef/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchCrate }:
```

File: pkgs/development/tools/rust/cargo-chef/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-d467uk4UCtAKcpFYODxIhRrYoIOHzxhoaJVMA9ErRAw=";
+  };
+
+  cargoSha256 = "sha256-5xj4/uxuMhlqY1ncrMU1IFWdVB4ZjHVXg0ZbRXDvIak=";
+
+  doCheck = true;
+
```
Comment (figsoda) [inline pkgs/development/tools/rust/cargo-chef/default.nix:15]:
```suggestion
```

File: pkgs/development/tools/rust/cargo-chef/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  meta = with lib; {
+    description = "Rust cargo dependencies cacher";
```
Comment (figsoda) [inline pkgs/development/tools/rust/cargo-chef/default.nix:17]:
```suggestion
    description = "A cargo-subcommand to speed up Rust Docker builds using Docker layer caching";
```

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -15358,6 +15358,7 @@ with pkgs;
   cargo-make = callPackage ../development/tools/rust/cargo-make {
     inherit (darwin.apple_sdk.frameworks) Security SystemConfiguration;
   };
+  cargo-chef = callPackage ../development/tools/rust/cargo-chef { };
```
Comment (figsoda) [inline pkgs/top-level/all-packages.nix:15361]: this should be in mostly alphabetical order

**Review by figsoda — CHANGES_REQUESTED** [review]

**Comment by kkharji** [comment]
@figsoda Awesome review. Could check now. Thanks

File: pkgs/development/tools/rust/cargo-chef/default.nix [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{ lib, rustPlatform, fetchCrate, }:
```
Comment (figsoda) [inline pkgs/development/tools/rust/cargo-chef/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchCrate }:
```

File: pkgs/development/tools/rust/cargo-chef/default.nix [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-d467uk4UCtAKcpFYODxIhRrYoIOHzxhoaJVMA9ErRAw=";
+  };
+
+  cargoSha256 = "sha256-5xj4/uxuMhlqY1ncrMU1IFWdVB4ZjHVXg0ZbRXDvIak=";
+
+  meta = with lib; {
+    description =
+      "A cargo-subcommand to speed up Rust Docker builds using Docker layer caching";
```
Comment (figsoda) [inline pkgs/development/tools/rust/cargo-chef/default.nix:16]:
```suggestion
    description = "A cargo-subcommand to speed up Rust Docker builds using Docker layer caching";
```

**Review by figsoda — APPROVED** [review]
lgtm, just some formatting suggestions

**Comment by Artturin** [comment]
fixup the Update commits

**Comment by kkharji** [comment]
> fixup the Update commits to the init commit

Done

**Review by figsoda — APPROVED** [review]
